business/workloads.go

```diff
@@ -2311,17 +2321,23 @@ func (in *WorkloadService) GetWaypointsForWorkload(ctx context.Context, workload
 
 	// then, get the waypoint workloads to filter out "forNone" waypoints
 	for _, waypoint := range waypoints {
-		waypointWorkload, err := in.fetchWorkload(ctx, WorkloadCriteria{Cluster: workload.Cluster, Namespace: waypoint.Namespace, WorkloadName: waypoint.Name, WorkloadGVK: schema.GroupVersionKind{}, IncludeWaypoints: false})
-		if err != nil {
-			log.Debugf("GetWaypointsForWorkload: Error fetching waypoint workload %s", err.Error())
-			return nil
-		}
-		waypointFor, waypointForFound := waypointWorkload.Labels[config.WaypointFor]
-		if !waypointForFound || waypointFor != config.WaypointForNone {
-			key := fmt.Sprintf("%s_%s_%s", workload.Cluster, waypoint.Namespace, waypoint.Name)
-			if waypointWorkload != nil && !workloadsMap[key] {
-				workloadsList = append(workloadsList, models.WorkloadReferenceInfo{Name: waypoint.Name, Namespace: waypoint.Namespace, Cluster: waypoint.Cluster, Type: waypointWorkload.WaypointFor()})
-				workloadsMap[key] = true
+		waypointWorkloads := in.GetWaypoints(ctx)
+		found := false
+		waypointWorkload := models.Workload{}
+		for _, ww := range waypointWorkloads {
+			if ww.Name == workload.Name && ww.Namespace == workload.Namespace && ww.Cluster == workload.Cluster {
```

Comment:
I think this line is incorrect.  I think you wanted to test against the capturing waypoint workload, not the "app" workload.  I changed it to:
`if ww.Name == waypoint.Name && ww.Namespace == waypoint.Namespace && ww.Cluster == waypoint.Cluster {`